src/Ast.ml

```diff
@@ -44,6 +44,23 @@ let is_prefix exp =
   | Pexp_ident {txt= Lident i; _} -> is_prefix_id i
   | _ -> false
 
+let is_infix_symbol = function
+  | '$' | '&' | '*' | '+' | '-' | '/' | '<' | '=' | '>' | '@' | '^' | '|' ->
+      true
+  | _ -> false
+
+let is_monadic_binding_id s =
+  String.length s > 3
+  && (String.is_prefix s ~prefix:"let" || String.is_prefix s ~prefix:"and")
+  && String.for_all
+       (String.sub s ~pos:3 ~len:(String.length s - 3))
```

Comment:
Can we find a way to not copy the string ?

Reply:
Using `lfindi` now, it should be better.